src/log/Log.cc

```diff
@@ -354,9 +354,9 @@ void Log::_flush(EntryQueue *t, EntryQueue *requeue, bool crash)
 void Log::_log_message(const char *s, bool crash)
 {
   if (m_fd >= 0) {
-    int r = safe_write(m_fd, s, strlen(s));
-    if (r >= 0)
-      r = safe_write(m_fd, "\n", 1);
+    std::string b(s);
+    b += '\n';
```

Comment:
the `std::string` constructor is likely to allocate only enough memory to fit `s`, so the `+=` will probably do an extra allocation

you could use `std::string::reserve()` to fit everything in one heap allocation, but instead of using the heap, maybe consider putting it on the stack with `alloca()`?